src/components/TreeView/TreeView.tsx

```diff
@@ -38,72 +36,68 @@ export function TreeView({
     additionalNodeElements,
     level,
 }: TreeViewProps) {
-    const rootRef = React.useRef<HTMLDivElement>(null);
-
-    const nodeLabel = (
-        <div className={b('content')}>
-            {icon && <div className={b('icon')}>{icon}</div>}
-            <div className={b('text')} title={title}>
-                {name}
-            </div>
-            {actions && actions.length > 0 && (
-                <div className={b('actions')}>
-                    {additionalNodeElements}
-                    <DropdownMenu
-                        defaultSwitcherProps={{
-                            view: 'flat-secondary',
-                            size: 's',
-                            pin: 'brick-brick',
-                        }}
-                        items={actions}
-                    />
-                </div>
-            )}
-        </div>
-    );
-
-    React.useEffect(() => {
-        const rootEl = rootRef.current;
-        const itemEl = rootEl && rootEl.querySelector<HTMLDivElement>(`.${b('item')}`);
-
-        if (!onClick || !itemEl) {
-            return;
-        }
-
-        const skipClicksSelector = `.tree-view_arrow, .${b('actions')}`;
+    const handleClick = useCallback<MouseEventHandler>(
+        (event) => {
+            if (!onClick) return;
 
-        function handleClick(event: MouseEvent) {
-            const path = event
+            const shouldSkip = event.nativeEvent
                 .composedPath()
-                .filter((item) => (item as Node).nodeType === Node.ELEMENT_NODE) as Element[];
-            const hasSkipped = path.some((el) => el.matches(skipClicksSelector));
+                .some(
+                    (target) =>
+                        target instanceof HTMLElement &&
+                        ((target.nodeName === 'BUTTON' && !target.hasAttribute('disabled')) ||
+                            (target.hasAttribute('tabindex') && target.tabIndex > -1)),
+                );
 
-            if (!hasSkipped) {
-                onClick!();
-            }
-        }
+            if (!shouldSkip) onClick();
+        },
+        [onClick],
+    );
 
-        itemEl.addEventListener('click', handleClick);
+    const handleArrowClick = onArrowClick || onClick;
 
-        return () => {
-            itemEl.removeEventListener('click', handleClick);
-        };
-    }, [onClick]);
+    let arrowClassName = 'tree-view_arrow';
+    let containerClassName = 'tree-view_children';
+    if (collapsed) {
+        arrowClassName += ' tree-view_arrow-collapsed';
+        containerClassName += ' tree-view_children-collapsed';
+    }
```

Comment:
Left as it was to avoid rewriting selectors in styles.

Reply:
Probably, it will be better to add additional `block('tree-view')` and set this styles with this block: `className={treeViewBlock('arrow', {collapsed})}` . With comment, that it was done not to override selectors

Reply:
As I see, `block` uses `__` as separator for elements and `react-treeview` uses `_`. I can use `block` like `treeViewBlock({arrow: true, 'arrow-collapsed': collapsed})` but it seems confusing.

I suggest just comment on what is happening with these `className`s.

Reply:
I see. Well, in this case, you could just comment, probably mentioning use cases, if someone further will want to update it:

https://github.com/ydb-platform/ydb-embedded-ui/blob/main/src/containers/Tenant/Diagnostics/TenantOverview/Healthcheck/IssuesViewer/IssueTree.scss

https://github.com/ydb-platform/ydb-embedded-ui/blob/main/src/containers/Versions/GroupedNodesTree/GroupedNodesTree.scss

Reply:
Done! Also rewrote `TreeView`\`s own selectors to BEM notation.